lab-traceyr/server.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const net = require('net');
+const ClientPool = require('./lib/clientPool');
+
+var clients = new ClientPool();
+
+module.exports = exports = net.createServer(function(socket){
+  clients.ee.emit('register', socket);
+
+  console.log(socket.nickname + ' has joined the network');
+
+  socket.on('error', function(err){
+    if(err) console.log(err);
+  });
+
+  socket.on('data', function(data){
+    if(data.toString() === 'END\r\n'){
+      socket.end();
+    } else {
+      clients.ee.emit('broadcast', data, socket);
+    }
+  });
+
+  socket.on('end', function(){
```

Comment:
Interesting. I wonder how you got socket.on('end') to work when Tyler had to use socket.on('close').